server/data/data-jmap-cassandra/src/main/java/org/apache/james/jmap/cassandra/filtering/RuleDTO.java

```diff
@@ -254,13 +279,21 @@ public ActionDTO getAction() {
     }
 
     public Rule toRule() {
-        return Rule.builder()
+        Rule.Builder ruleBuilder = Rule.builder()
             .id(Rule.Id.of(id))
             .name(name)
-            .condition(conditionDTO.toCondition())
-            .name(name)
-            .action(actionDTO.toAction())
-            .build();
+            .action(actionDTO.toAction());
+
+        if (conditionDTO != null) {
+            ruleBuilder.conditions(Arrays.asList(conditionDTO.toCondition()));
+        } else {
+            ruleBuilder.conditions(conditionDTOs.stream().map(ConditionDTO::toCondition).collect(Collectors.toList()));
+        }
+
+        if (conditionCombiner != null) ruleBuilder.conditionCombiner(conditionCombiner);
+        else ruleBuilder.conditionCombiner(Rule.ConditionCombiner.AND);
```

Comment:
No if else blocks like this, should be:
```
if (condition) {
    code
} else {
    code
}
```
better readability

Reply:
I think we could have kept the initial layout? Could extract the if else condition to functions